pueue_lib/src/settings.rs

```diff
@@ -46,6 +46,12 @@ pub struct Shared {
     /// The path to the unix socket.
     #[cfg(not(target_os = "windows"))]
     pub unix_socket_path: Option<PathBuf>,
+    /// Unix socket permissions. Typically specified as an octal number and
+    /// defaults to `0o700` which grants only the current user access to the
+    /// socket. For a client to connect to the daemon, the client must have
+    /// read/write permissions.
+    #[cfg(not(target_os = "windows"))]
+    pub unix_socket_permissions: Option<nix::sys::stat::mode_t>,
```

Comment:
Change to `u32`. `mode_t` is just a type alias:
https://docs.rs/nix/latest/nix/sys/stat/type.mode_t.html

Reply:
If the `nix` dependency is removed, then this point is moot. 

Otherwise, the `mode_t` is still required because the `mode_t` is a different alias on different machines. On macOS for example, `mode_t` is an alias for `u16`.